prime-router/src/main/kotlin/tokens/OktaAuthentication.kt

```diff
@@ -2,53 +2,122 @@ package gov.cdc.prime.router.tokens
+     *
+     * Return the claims found in the jwt if the jwt token in [request] is validated.
+     * Return null if not authenticated.
+     * Always performs authentication using Okta, unless running locally.
+     */
+    fun authenticate(request: HttpRequestMessage<String?>): AuthenticatedClaims? {
+        val accessToken = getAccessToken(request)
+        return authenticate(accessToken, request.httpMethod, request.uri.path)
+    }
 
-        fun setRequest(request: HttpRequestMessage<String?>) {
-            httpRequestMessage = request
+    /**
+     * See full comments above.
+     */
+    fun authenticate(
+        accessToken: String?,
+        httpMethod: HttpMethod,
+        path: String,
+    ): AuthenticatedClaims? {
+        if (isLocal(accessToken)) {
+            logger.info("Granted test auth request for $httpMethod:$path")
+            return AuthenticatedClaims.generateTestClaims()
         }
 
-        fun authenticationVerifier(): AuthenticationVerifier {
-            // If we are running this locally, use the TestAuthenticationVerifier
-            // To test locally _with_ auth, add a 'localauth=true' header to your POST.
-            val localAuth = httpRequestMessage?.headers?.get("localauth")
+        // Confirm the token exists.
+        if (accessToken == null) {
+            logger.info("Missing Authorization Header: $httpMethod:$path}")
+            return null
+        }
 
-            return if (!Environment.isLocal()) {
-                OktaAuthenticationVerifier()
-            } else if (localAuth != null && localAuth == "true") {
-                OktaAuthenticationVerifier()
-            } else {
-                logger.info("No auth needed - running locally")
-                TestAuthenticationVerifier()
-            }
+        try {
+            val jwtVerifier = JwtVerifiers.accessTokenVerifierBuilder()
+                .setIssuer("https://$issuerBaseUrl/oauth2/default")
+                .build()
+            // Perform authentication.  Throws exception if authentication fails.
+            val jwt = jwtVerifier.decode(accessToken)
+
+            // Extract claims into a more usable form
+            val claims = AuthenticatedClaims(jwt.claims)
+            logger.info("Authenticated request by ${claims.userName}: $httpMethod:$path")
+            return claims
+        } catch (e: JwtVerificationException) {
+            logger.info("JWT token failed to authenticate for call: $httpMethod: $path", e)
+            return null
+        } catch (e: Exception) {
+            logger.info("Failure while authenticating, for call: $httpMethod: $path", e)
+            return null
+        }
+    }
+
+    /**
+     * Helper method for authentication.
+     * Check whether we are running locally
+     */
+    fun isLocal(accessToken: String?): Boolean {
+        return if (!Environment.isLocal()) {
+            false
+        } else if (accessToken != null && accessToken.split(".").size == 3) {
```

Comment:
How about if we just test for `accessToken != null`?  If there is any token then use the parser.  I get null when no Authorization header is set.  I get an empty string (`""`) when no string is added after the Bearer keyword.

Reply:
I looked into this, but it broke quite a number of unit tests, so for now I'll defer this as separate work.